kicost/eda_tools/generic_csv/generic_csv.py

```diff
@@ -85,23 +79,21 @@ def get_part_groups(in_file, ignore_fields, variant):
     # If the first line contains a column header that is not in the list of
     # allowable field names, then assume the first line is data and not a header.
     field_names = list(field_name_translations.keys()) + list(field_name_translations.values())
-    for col_hdr in header:
-        if col_hdr not in field_names:
-            # If a column header is not in the list of field names, then there is
-            # no header in the file. Therefore, create a header based on number of columns.
- 
-            # header may have a '' at the end, so remove it.
-            if '' in header:
-                header.remove('')
-
-            num_cols = len(header)
-            if num_cols == 1:
-                header = ['manf#']
-            elif num_cols == 2:
-                header = ['manf#', 'refs']
-            else:
-                header = ['qty', 'manf#', 'refs']
-            break
+    if not any(col_hdr.lower() in field_names for col_hdr in header):

```

Comment:
I think this should be:

    if any(col_hdr.lower() not in field_names for col_hdr in header):

Because the original code says there is no header if **any** of the column headers is not in the list of field names. The new code says there is no header only if **all** the column headers are not in the list of field names.

Reply:
Yes, maybe is needing some modification. But with `if any(col_hdr.lower() not in field_names for col_hdr in header):`, if there is any header that is not in the dictionary (keys or values) so it will be assumed the default header sequence.
This may create some problem if I use `manf` (to manufacturer producer for example) or `desc` (for description in the spreadsheet).

If `if not any(col_hdr.lower() in field_names for col_hdr in header):` the default is assumed if it is not founded any of the headers (listed in the dict()). Maybe could be improved to:

    if not col_hdr.lower() not ('manf#'):

?
Due `manf#` is the most important header titlte to be founded.

When I created this module I thought:
1) Have to found `manf#`, even if the quantity is not specified (so "assume one to board" just to create the spreadsheet and scrap the web pages);
2) If founded `manf#` and `qty` but not the `refs`, create the `refs` by `GENERIC_PREFIX`;
3) If founded `manf#` and `refs` use the `split_refs()` to create the splited one-by-one designator and ignore `qty`, if present;
4) Others headers could be present, such as `values`, `desc`, `manf`... and be used KiCost (even Lib or fooprint/package, that are assume `???` when not founded).

Reply:
I did some modifications and tested in the CSV files in the "forum".